File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/datasource/AccountDataSource.java
package io.github.skywalkerdarren.simpleaccounting.model.datasource;

import java.util.List;
import java.util.UUID;

import io.github.skywalkerdarren.simpleaccounting.model.entity.Account;

public interface AccountDataSource {


    void getAccount(UUID uuid, LoadAccountCallBack callBack);

    void getAccounts(LoadAccountsCallBack callBack);

    void delAccount(UUID uuid);

    void changePosition(Account a, Account b);

    interface LoadAccountCallBack {
        void onAccountLoaded(Account account);
    }

    interface LoadAccountsCallBack {
        void onAccountsLoaded(List<Account> accounts);
    }
}

File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/datasource/AccountDataSource.kt
package io.github.skywalkerdarren.simpleaccounting.model.datasource

import io.github.skywalkerdarren.simpleaccounting.model.entity.Account
import java.math.BigDecimal
import java.util.*

interface AccountDataSource {
    fun getAccount(uuid: UUID, callBack: LoadAccountCallBack)
    fun getAccounts(callBack: LoadAccountsCallBack)
    fun delAccount(uuid: UUID)
    fun changePosition(a: Account, b: Account)
    fun updateAccountBalance(uuid: UUID, balance: BigDecimal)

    interface LoadAccountCallBack {
        fun onAccountLoaded(account: Account)
    }

    interface LoadAccountsCallBack {
        fun onAccountsLoaded(accounts: List<Account>?)
    }
}